src/viam/app/data_client.py

```diff
@@ -184,47 +188,65 @@ def __init__(self, channel: Channel, metadata: Mapping[str, str]):
+    ) -> Tuple[List[TabularData], int, str]:
+        """Filter and download tabular data. The data will be paginated into pages of `limit` items, and the pagination ID will be included
+        in the returned tuple. If a destination is provided, the data will be saved to that file.
+        If the file is not empty, it will be overwritten.
 
         ::
 
             from viam.proto.app.data import Filter
 
+            my_data = []
+            last = None
             my_filter = Filter(component_name="left_motor")
-            tabular_data = await data_client.tabular_data_by_filter(my_filter)
+            while True:
+                tabular_data, last = await data_client.tabular_data_by_filter(my_filter, last)
+                if not tabular_data:
+                    break
+                my_data.extend(tabular_data)
+
 
         Args:
             filter (viam.proto.app.data.Filter): Optional `Filter` specifying tabular data to retrieve. No `Filter` implies all tabular
                 data.
+            limit (int): The maximum number of entries to include in a page. Defaults to 50.
+            sort_order (viam.proto.app.data.Order): The desired sort order of the data.
+            last (str): Optional string indicating the ID of the last-returned data.
+                        If provided, the server will return the next data entries after the `last` ID.
+            count_only (bool): Whether to return only the total count of entries.
             dest (str): Optional filepath for writing retrieved data.
 
         Returns:
             List[TabularData]: The tabular data.
+            int: The count (number of entries)
+            str: The last-returned page ID.
         """
         filter = filter if filter else Filter()
-        last = ""
-        data = []
-
-        # `DataRequest`s are limited to 100 pieces of data, so we loop through calls until
-        # we are certain we've received everything.
-        while True:
-            data_request = DataRequest(filter=filter, limit=100, last=last)
-            request = TabularDataByFilterRequest(data_request=data_request, count_only=False)
-            response: TabularDataByFilterResponse = await self._data_client.TabularDataByFilter(request, metadata=self._metadata)
-            if not response.data or len(response.data) == 0:
-                break
-            data += [
-                DataClient.TabularData(
-                    struct_to_dict(struct.data),
-                    response.metadata[struct.metadata_index],
-                    struct.time_requested.ToDatetime(),
-                    struct.time_received.ToDatetime(),
-                )
-                for struct in response.data
-            ]
-            last = response.last
+
+        data_request = DataRequest(filter=filter)
+        if limit:
+            data_request.limit = limit
+        if sort_order:
+            data_request.sort_order = sort_order
+        if last:
```

Comment:
Awesome, thanks for updating here and in BinaryDataByFilter! Similar to my PR-level question about testing, could we confirm that keeping these unspecified gives back what we expect in an SDK call?

Reply:
I haven't manually tested this code -- I don't have a robot with a ton of data. But also, this is just client code. I added tests to make sure that the client is forwarding values properly to the server, but beyond that the client is simply going to trust that the server returns back the appropriate data for the request

Reply:
Got it, so confirming that if these are unspecified, the proto field is unspecified. If so, that matches what we'd want.

Reply:
LGTMed, but I just added you into the Data/ML Dev Org so that you can test against this, just so we can be extra sure that this is following the behavior we expect before giving customer updates.